src/main/java/io/druid/utils/CompressionUtils.java

```diff
@@ -26,82 +13,46 @@
+  @Deprecated // Use com.metamx.common.CompressionUtils.zip
   public static long zip(File directory, OutputStream out) throws IOException
   {
-    if (!directory.isDirectory()) {
-      throw new IOException(String.format("directory[%s] is not a directory", directory));
-    }
-
-    long totalSize = 0;
-    ZipOutputStream zipOut = null;
-    try {
-      zipOut = new ZipOutputStream(out);
-      File[] files = directory.listFiles();
-      for (File file : files) {
-        log.info("Adding file[%s] with size[%,d].  Total size so far[%,d]", file, file.length(), totalSize);
-        if (file.length() >= Integer.MAX_VALUE) {
-          zipOut.finish();
-          throw new IOException(String.format("file[%s] too large [%,d]", file, file.length()));
-        }
-        zipOut.putNextEntry(new ZipEntry(file.getName()));
-        totalSize += ByteStreams.copy(Files.newInputStreamSupplier(file), zipOut);
-      }
-      zipOut.closeEntry();
-    }
-    finally {
-      if (zipOut != null) {
-        zipOut.finish();
-      }
-    }
-
-    return totalSize;
+    return com.metamx.common.CompressionUtils.zip(directory, out);
   }
 
+  @Deprecated // Use com.metamx.common.CompressionUtils.unzip
   public static void unzip(File pulledFile, File outDir) throws IOException
   {
-    if (!(outDir.exists() && outDir.isDirectory())) {
-      throw new ISE("outDir[%s] must exist and be a directory", outDir);
-    }
-
-    log.info("Unzipping file[%s] to [%s]", pulledFile, outDir);
-    InputStream in = null;
-    try {
-      in = new BufferedInputStream(new FileInputStream(pulledFile));
-      unzip(in, outDir);
-    }
-    finally {
-      CloseQuietly.close(in);
-    }
+    com.metamx.common.CompressionUtils.unzip(pulledFile, outDir);
   }
 
+  @Deprecated // Use com.metamx.common.CompressionUtils.unzip
   public static void unzip(InputStream in, File outDir) throws IOException
   {
-    ZipInputStream zipIn = new ZipInputStream(in);
-
-    ZipEntry entry;
-    while ((entry = zipIn.getNextEntry()) != null) {
-      try (FileOutputStream out = new FileOutputStream(new File(outDir, entry.getName()))){
-        ByteStreams.copy(zipIn, out);
-        zipIn.closeEntry();
-      }
-    }
+    com.metamx.common.CompressionUtils.unzip(in, outDir);
   }
 
+  /**
+   * Uncompress using a gzip uncompress algorithm from the `pulledFile` to the `outDir`.
+   * Unlike `com.metamx.common.CompressionUtils.gunzip`, this function takes an output *DIRECTORY* and tries to guess the file name.
+   * It is recommended that the caller use `com.metamx.common.CompressionUtils.gunzip` and specify the output file themselves to ensure names are as expected
+   *
+   * @param pulledFile The source file
+   * @param outDir     The destination directory to put the resulting file
+   *
+   * @throws IOException on propogated IO exception, IAE if it cannot determine the proper new name for `pulledFile`
+   */
   public static void gunzip(File pulledFile, File outDir) throws IOException
```

Comment:
Because the functionality is slightly different (outDir vs outFile) and there is no alternative for specifying an output directory yet.